[JIT] add support for torch.cdist

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33818 [JIT] Add support for list()
* #33783 [JIT] add support for torch.norm
* **#33737 [JIT] add support for torch.cdist**
* #33736 [JIT] add support for lu_unpack
* #33724 [JIT] add support for torch.lu to torchscript
* #33504 [JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft

Differential Revision: [D20121916](https://our.internmc.facebook.com/intern/diff/D20121916)

Commit:
[JIT] add support for torch.cdist

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.cdist"




[ghstack-poisoned]

vishwakftw reviewed (approved):
Looks good to me, thank you.

vishwakftw commented:
Are the docs failures normal?

eellison commented:
> Are the docs failures normal?

i think i just need to rename `indices_product` to `_indices_product` the PR below this

eellison commented on torch/jit/_builtins.py:
```diff
@@ -83,7 +83,8 @@
     (torch._C._get_tracing_state, "aten::_get_tracing_state"),
     (warnings.warn, "aten::warn"),
-    (torch._VF.stft, "aten::stft")
+    (torch._VF.stft, "aten::stft"),
```
i think it's something to do with native schema bound ops name collision.

Commit:
Update on "[JIT] add support for torch.cdist"




[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.cdist"




[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.cdist"


Differential Revision: [D20121916](https://our.internmc.facebook.com/intern/diff/D20121916)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.cdist"


Differential Revision: [D20121916](https://our.internmc.facebook.com/intern/diff/D20121916)

[ghstack-poisoned]
